express-mutations/server.ts

```diff
@@ -0,0 +1,126 @@
+import express from 'express';
+import pg from 'pg';
+import { ClientError, errorMiddleware } from './lib/index.js';
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/pagila',
+  ssl: {
+    // Allow non-SSL traffic to localhost
+    rejectUnauthorized: false,
+  },
+});
+
+const app = express();
+app.use(express.json());
+
+// Endpoint for testing
+app.get('/api/actors/:actorId', async (req, res, next) => {
+  try {
+    const { actorId } = req.params;
+    if (!Number.isInteger(+actorId)) {
+      throw new ClientError(400, `Non-integer actorId: ${actorId}`);
+    }
+    const sql = `
+      select * from "actors"
+      where "actorId" = $1;
+    `;
+    const params = [actorId];
+    const result = await db.query(sql, params);
+    const actor = result.rows[0];
+    if (!actor) throw new ClientError(404, `actor ${actorId} not found`);
+    res.json(actor);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.post('/api/actors', async (req, res, next) => {
+  try {
+    const sql = `
+      insert into "actors" ("firstName", "lastName")
+      values ($1, $2)
+      returning *;
+    `;
+    // console.log('req.body', req.body)
+    const params = [req.body.firstName, req.body.lastName];
+    if (!req.body.firstName) {
+      throw new ClientError(404, 'Please provide a first name.');
+    }
+    if (!req.body.lastName) {
+      throw new ClientError(404, 'Please provide a last name.');
+    }
+    const { rows } = await db.query(sql, params);
+    const [actor] = rows;
+    if (!actor) {
```

Comment:
This seems like a good thing to check, but it turns out that an `insert` statement will either always succeed or it will throw an exception. So this check will actually never fire.